tests/unit/test_cli.py

```diff
@@ -207,130 +221,170 @@ def test_specify_packages(
                 ],
             ),
             (
-                ['--package={"type": "gomod", "path": "no-such-dir"}'],
+                '{"type": "gomod", "path": "no-such-dir"}',
                 [
                     "1 validation error for user input",
                     "packages -> 0",
                     "package path does not exist (or is not a directory): no-such-dir",
                 ],
             ),
+            # Extra fields
             (
-                ['--package={"type": "gomod", "what": "dunno"}'],
+                '{"type": "gomod", "what": "dunno"}',
                 [
                     "1 validation error for user input",
                     "packages -> 0 -> GomodPackageInput -> what",
                     "extra fields not permitted",
                 ],
             ),
+            # Invalid format using 'packages' key
+            (
+                '{"packages": "gomod"}',
+                [
+                    "1 validation error for user input",
+                    "packages",
+                    "value is not a valid list",
+                ],
+            ),
+            (
+                '{"packages": {"type":"gomod"}}',
+                [
+                    "1 validation error for user input",
+                    "packages",
+                    "value is not a valid list",
+                ],
+            ),
+            (
+                '{"packages": ["gomod"]}',
+                [
+                    "1 validation error for user input",
+                    "packages -> 0",
+                    "Discriminator 'type' is missing",
+                ],
+            ),
         ],
     )
     def test_invalid_packages(
-        self, package_args: list[str], expect_error_lines: list[str], tmp_cwd: Path
+        self, package_args: str, expect_error_lines: list[str], tmp_cwd: Path
     ):
         tmp_cwd.joinpath("suspicious-symlink").symlink_to("..")
 
-        result = invoke_expecting_invalid_usage(app, ["fetch-deps", *package_args])
+        result = invoke_expecting_invalid_usage(app, ["fetch-deps", package_args])
 
         for pattern in expect_error_lines:
             assert_pattern_in_output(pattern, result.output)
 
     @pytest.mark.parametrize(
         "flag_args, expect_flags",
         [
-            ([], {}),
-            (["--gomod-vendor"], {"gomod-vendor"}),
-            (["--flags=gomod-vendor"], {"gomod-vendor"}),
-            (["--gomod-vendor", "--flags=gomod-vendor"], {"gomod-vendor"}),
+            (["gomod"], {}),
+            (["gomod", "--gomod-vendor"], {"gomod-vendor"}),
+            (['{"packages": [{"type":"gomod"}], "flags": "gomod-vendor"}'], {"gomod-vendor"}),
+            (['{"packages": [{"type":"gomod"}], "flags": ["gomod-vendor"]}'], {"gomod-vendor"}),
+            (
+                ['{"packages": [{"type":"gomod"}], "flags": ["gomod-vendor"]}', "--gomod-vendor"],
+                {"gomod-vendor"},
```

Comment:
We don't seem to have a test where we actually combine `PKG` flags and `--flag` flags, this one only tests de-duplication (we didn't even have one before /o\\)

Reply:
I added a single scenario to cover combining flags... let me know if you think it's enough.